pkg/collector/metric_collector.go

```diff
@@ -112,7 +113,10 @@ func (c *Collector) Update() {
 	c.updateBPFMetrics() // collect new hardware counter metrics if possible
 
 	// TODO: collect cgroup metrics only from cgroup to avoid unnecessary overhead to kubelet
-	c.updateCgroupMetrics()  // collect new cgroup metrics from cgroup
+	// collect cgroup metrics only if no power meter is available or if cgroup metrics are exported
+	if !components.IsSystemCollectionSupported() || config.ExposeCgroupMetrics {
```

Comment:
can we store this `!components.IsSystemCollectionSupported` into a bool so we won't need get this all the time?
seems this call in some implementation is expensive